lib/src/repo_path.rs

```diff
@@ -468,6 +468,77 @@ fn is_valid_repo_path_str(value: &str) -> bool {
     !value.starts_with('/') && !value.ends_with('/') && !value.contains("//")
 }
 
+/// An error from `RepoPathUiConverter::parse_file_path`.
+#[derive(Debug, Error)]
+#[error(r#"Invalid path "{input}""#)]
+pub struct UiPathParseError {
+    /// Input path.
+    pub input: String,
+    /// Source error.
+    pub source: Box<dyn std::error::Error + Send + Sync>,
```

Comment:
nit: I think this error type can be a transparent wrapper of `Box<dyn Error>`. It doesn't add any useful information to `FsPathParseError`.

Reply:
We use the `input` field [here](https://github.com/martinvonz/jj/blob/b31f75bc94c4552454ab929b54b9f37340231fac/cli/src/command_error.rs#L556-L558). I suppose we could manually implement `Display` and `Error` on this type. I'm guessing it's not worth it, but let me know if you think we should (or if you have another solution).